aidial_adapter_openai/legacy_completions.py

```diff
@@ -0,0 +1,102 @@
+from aidial_sdk.exceptions import HTTPException as DialException
+from fastapi import Response
+from fastapi.responses import StreamingResponse
+from openai import Completion
+from openai.openai_object import OpenAIObject
+
+from aidial_adapter_openai.constant import DEFAULT_TIMEOUT
+from aidial_adapter_openai.env import COMPLETION_DEPLOYMENTS_PROMPT_TEMPLATES
+from aidial_adapter_openai.utils.auth import get_auth_header
+from aidial_adapter_openai.utils.exceptions import handle_exceptions
+from aidial_adapter_openai.utils.sse_stream import to_openai_sse_stream
+from aidial_adapter_openai.utils.streaming import map_stream
+
+
+def convert_completion_to_chat_completion(item: OpenAIObject, role="assistant"):
+    item_dict = item.to_dict_recursive()
+    return {
+        "object": "chat.completion",
+        "id": item_dict["id"],
+        "model": item_dict["model"],
+        "created": item_dict["created"],
+        "usage": item_dict["usage"],
+        "choices": [
+            {
+                "index": choice["index"],
+                "finish_reason": choice["finish_reason"],
+                "message": {
+                    "content": choice["text"],
+                    "role": role,
+                },
+                "logprobs": choice["logprobs"],
+            }
+            for choice in item_dict["choices"]
+        ],
+    }
+
+
+async def legacy_completions(
+    data,
```

Comment:
Let's add type hints

Reply:
fixed, but adding type hints to data cause adding typing ignore somewhere, because linter thinks that

`data.get('n', 1)` still can be None

Reply:
`data: dict` is enough :)

I want all of the parameters to be typed to a reasonable extent, which is the case for the rest of the project.

Reply:
I think, that such `data: dict` type hints are mostly useless, other than preventing some really basic errors.
It's the same as adding `data: object` or `data: Any`.

Even when I know that data is `dict/list/object`, I have no idea at all, how to actually work with this parameter inside of code.

Reply:
I understand that it's hard to add specific types to all nested objects, but at least we can do it in some places, to make life of developer easier, and prevent him from making `print(data)` to actually understand, what's hidden inside of this dictionary